server/chat-plugins/helptickets-auto.ts

```diff
@@ -0,0 +1,671 @@
+import {FS, Utils} from '../../lib';
+import type {ModlogSearch, ModlogEntry} from '../modlog';
+import {
+	TicketState, getBattleLog, getBattleLinks, writeTickets, notifyStaff, writeStats, HelpTicket,
+} from './helptickets';
+import * as Artemis from '../artemis';
+
+const ORDERED_PUNISHMENTS = ['WARN', 'FORCERENAME', 'LOCK', 'NAMELOCK', 'WEEKLOCK', 'WEEKNAMELOCK'];
+const PMLOG_IGNORE_TIME = 24 * 60 * 60 * 1000;
+const WHITELIST = ['mia'];
+
+export interface AutoPunishment {
+	modlogCount?: number;
+	severity?: {type: string[], certainty: number};
+	/**
+	 * Should it be run on just one message?
+	 * or is it safe to run on averages (for the types that use averages)
+	 */
+	isSingleMessage?: boolean;
+	punishment: string;
+	ticketType: string;
+}
+
+export interface AutoSettings {
+	punishments: AutoPunishment[];
+	applyPunishments: boolean;
+}
+
+const defaults: AutoSettings = {
+	punishments: [{
+		ticketType: 'inapname',
+		punishment: 'forcerename',
+		severity: {type: ['sexual_explicit', 'severe_toxicity', 'identity_attack'], certainty: 0.4},
+	}, {
+		ticketType: 'pmharassment',
+		punishment: 'warn',
+		severity: {type: ['sexual_explicit', 'severe_toxicity', 'identity_attack'], certainty: 0.15},
+	}],
+	applyPunishments: false,
+};
+
+export const settings: AutoSettings = (() => {
+	try {
+		// spreading w/ default means that
+		// adding new things won't crash by not existing
+		return {...defaults, ...JSON.parse(FS('config/chat-plugins/ht-auto.json').readSync())};
+	} catch {
+		return defaults;
+	}
+})();
+
+function saveSettings() {
+	return FS('config/chat-plugins/ht-auto.json').writeUpdate(() => JSON.stringify(settings));
+}
+
+// partial is just to accept the 'partial' autopunishments done when they're created
+// (they aren't really partial but even if they are, it doesn't matter)
```

Comment:
I'm not sure what this comment refers to.

Reply:
Oh, I forgot to remove the comment when I removed the type.